### ISSUE TITLE
Run the bundler when the Gemfile changes

### ISSUE THREAD

**Issue by carllerche** [wycats/bundler#78]
Use mtime to see if bundler needs to run

**Comment by jeremy** [comment]
think this should use a SHA of the file contents rather than mtime

**Comment by thewoolleyman** [comment]
Yes, this is especially needed on CI environments where you don't want to have things fail and then go run bundler manually.

Here's what I want:
1. bundler always writes the SHA of the Gemfile in a file next to environment.rb.
2. There is a Bundler.uptodate? method which is a fast check whether the SHA is outdated.
3. You can then use Bundler.uptodate? in preinitializer.rb to conditionally run bundler only when it is needed.  Maybe even Bundler.run_if_needed or something.
